### PR TITLE
2nd Draft of Instructor Documentation

### PR DESCRIPTION
**Pull request by Nuyb**
Changed documentation format from CRUD to use cases.

### CODE REVIEW

File: documentation/docs/instructor.md [hunk 1]
```diff
@@ -1,82 +1,99 @@
...
 Example:
 ```
-American/New_York
+America/New_York
 ```
 
-## Hostname (Linux Only)
-Set the system hostname to be different than default value.
+### Hostname (Linux Only)
+Set a hostname that will replace the hostname generated by default.
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:68]: Explain the format of hostname generated by default.

File: documentation/docs/instructor.md [hunk 2]
```diff
@@ -1,82 +1,99 @@
...
+### Perform Reboot if Required After Package Install/Upgrade (Linux Only)
+By selecting yes, it will perform a reboot if packages are installed and upgraded.
+
+### Timezone (Linux Only)
+Set a different system timezone. 
+
 Example:
 ```
-American/New_York
+America/New_York
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:64]: Give more examples and link to a list of all time zones.

File: documentation/docs/instructor.md [hunk 3]
```diff
@@ -1,82 +1,99 @@
...
 ```
 
-## Hostname (Linux Only)
-Set the system hostname to be different than default value.
+### Hostname (Linux Only)
+Set a hostname that will replace the hostname generated by default.
+
+### Default User Password (Linux Only)
+Sets the password for default user (username is OS-dependant). Leave blank to disable.
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:71]: Link to documentation showing what is the username.
https://docs.cci.drexel.edu/display/cloud/Supported+Images (don't link to that page)

File: documentation/docs/instructor.md [hunk 4]
```diff
@@ -1,82 +1,99 @@
...
+
+### Default User Public Key (Linux Only)
+SSH public key for default user (username is OS-dependant).
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:74]: What is a default user?

File: documentation/docs/instructor.md [hunk 5]
```diff
@@ -1,82 +1,99 @@
...
+
+Example:
+```
+ssh-rsa AAAB... user@host
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:78]: It's not a full example, give examples of multiple types of public keys. Full ones, the dots are obviously standing for majority of characters.

File: documentation/docs/instructor.md [hunk 6]
```diff
@@ -1,82 +1,99 @@
...
+### Image 
+Choose the type of image you wish to use: CentOS 7.3, Ubuntu Server 16.04.2, Windows Server 2012 R2
+### Flavor
+Select the type of flavor: 01c-01m-10d reads 1 VCPU, 1 RAM GB, and 10 GB Total/Root Disk.
+### Shell Script
+The instance will run the script you provide here once it's created. Provide the appropriate shebang respective to the image type:
+
+Windows:
+```
+#ps1_sysnative
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:39]: What language will that be?

File: documentation/docs/instructor.md [hunk 7]
```diff
@@ -1,82 +1,99 @@
...
+
+`You can create multiple Environment Definitions for one Assignment, which can hold one or more VM Definitions.`
+
+## Creating a VM Definition
+By clicking on an environment definition title from the Environment Definition List page, you can view, create, edit, or delete  the Virtual Machine Definitions that have been created. Create a new VM by clicking 'Create VM Definition` in the top right corner. These VM Definitions will be used to load virtual machines into the environment when it's launched through an assignment. Find more details about specific fields in the creation of a VM below:  
+
+### Image 
+Choose the type of image you wish to use: CentOS 7.3, Ubuntu Server 16.04.2, Windows Server 2012 R2
+### Flavor
+Select the type of flavor: 01c-01m-10d reads 1 VCPU, 1 RAM GB, and 10 GB Total/Root Disk.
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:33]: That's wrong, this only applies to our cloud, other clouds may have completely different flavors and flavor names.

File: documentation/docs/instructor.md [hunk 8]
```diff
@@ -1,82 +1,99 @@
...
+
+`You can create multiple Environment Definitions for one Assignment, which can hold one or more VM Definitions.`
+
+## Creating a VM Definition
+By clicking on an environment definition title from the Environment Definition List page, you can view, create, edit, or delete  the Virtual Machine Definitions that have been created. Create a new VM by clicking 'Create VM Definition` in the top right corner. These VM Definitions will be used to load virtual machines into the environment when it's launched through an assignment. Find more details about specific fields in the creation of a VM below:  
+
+### Image 
+Choose the type of image you wish to use: CentOS 7.3, Ubuntu Server 16.04.2, Windows Server 2012 R2
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:31]: This is horribly wrong again, there is no guarantee such images exist.

File: documentation/docs/instructor.md [hunk 9]
```diff
@@ -1,82 +1,99 @@
...
+
+`Reminder: Your allocated quota is shared among all your students in all your courses.`
+
+## Creating a Course 
+Visit the Course List page by selecting the `Courses` tab found in the header. Create a VCL course by finding and clicking the `Create Course` button near the top right corner, providing a name for the desired course, and pressing `Submit`. Courses can be edited or deleted on the Course List page. 
+
+## Adding Students to a Course (CSV Upload)
+Click on a course title on the Course List page and then press the `Add Students (CSV Upload)` button. Upload a CSV file with proper fields `(Last Name, First Name, Username)`. Once uploaded, the data will be parsed and saved to the Cloud VCL database and the students listed in the CSV will be added to the course.
+
+`Reminder: The format of the CSV file used by Blackboard Learn may have additional fields; however, Cloud VCL only requires (Last Name, First Name, Username)`.
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:20]: GENERATED not used by BB Learn. WE use it!

File: documentation/docs/instructor.md [hunk 10]
```diff
@@ -1,82 +1,99 @@
...
+
+`Course and Environment Definition must first be created in order for students to be assigned to an Assignment.` 
+
+## Entering Your Environment
+On the Assignment List page, click on an assignment title and then press the `Enter Your Environment` button (it will take a moment for the console to load). Once the console appears and reaches the log in screen, enter the username and password given in the blue box. The console can be in full screen mode by clicking the `Open Console in Full Screen Mode`. If there are multiple VMs, you can easily switch from one console to another by clicking on the desired VM found in the green box. 
+
+`Please note: CTRL+C/CTRL+V doesn't work in this console.`
```
Comment (DSpeichert) [inline documentation/docs/instructor.md:95]: What else doesn't work? What is CTRL+C/CTRL+V?

**Review by DSpeichert — CHANGES_REQUESTED** [review]
Work more on this, my comments do not point out everything that is wrong with that.

File: documentation/docs/instructor.md [hunk 11]
```diff
@@ -1,82 +1,99 @@
...
-1. Go to the Assignment List page by selecting the `Assignment` tab found in the header.
-2. Find the assignment you want to delete and click on `Delete`.
-
-## Timezone (Linux Only)
-Set the system timezone for an instance.
+## Becoming an Instructor User
+Contact your system administrator to have them promote you to become an instructor user.
+
+## Logging Into Cloud VCL
+Navigate to https://cloudvcl.cci.drexel.edu/ and log in using your Drexel credentials. You will know that you've successfully logged in when you see `Logged in as YourUsername` in the top right corner. Once you log in and arrive on the home page, if you see a blue text box that indicates you're an instructor, you've been successfully promoted. If not, please ask your system administrator to promote you. To log out, select the `Logout` button in the top right corner.
```
Comment (abhishekvalanju) [inline documentation/docs/instructor.md:7]: Again, make this non-Drexel specific. Say instead, log in using your institution's credentials

File: documentation/docs/instructor.md (hunk 10 above)
Comment (Nuyb) [reply documentation/docs/instructor.md:95]: i'm not sure about what else doesn't work here

Comment (DSpeichert) [reply documentation/docs/instructor.md:95]: It's easier to say what works. Keyboard, Video, Mouse. Hence the name is KVM. So audio won't work.

**Pull request closed without merging**